osh/client/commands/common.py

```diff
@@ -150,3 +154,24 @@ def add_tarball_option(parser):
              "This option sets command which should build the package, usually this should be just "
              "\"make\", in case of packages which doesn't need to be built, just pass \"true\".",
     )
+
+
+def parse_json_option(option, opt, value, parser):
+    try:
+        json_value = json.loads(value)
+    except json.JSONDecodeError as e:
+        raise OptionValueError(f"Option {opt}: invalid JSON value: {e}")
```

Comment:
This is a good idea but the error message looks a bit too convoluted:
```console
$ osh-cli mock-build --metadata=
Usage: osh-cli mock-build [options] <args>

osh-cli: error: Option --metadata: invalid JSON value: Expecting value: line 1 column 1 (char 0)
```

Reply:
I've updated the exception to print the original json string instead. Thanks for suggestion!